depmapomics/upload.py

```diff
@@ -0,0 +1,613 @@
+        # ranking in source_priority
+        elif len(prs_in_mc[prs_in_mc.Datatype == "rna"]) > 1:
+            cds_ids = prs_in_mc[prs_in_mc.Datatype == "rna"].CDSID.tolist()
+            # at this point it is guaranteed that all cds_ids have different sources
+            subset_seq_table = seq_table[seq_table.index.isin(cds_ids)]
+            subset_seq_table.source = subset_seq_table.source.replace(source_priority)
+            latest_cds_id = subset_seq_table.loc[cds_ids, "source"].idxmin()
+            pr = subset_pr_table[subset_pr_table.CDSID == latest_cds_id].index[0]
+            rows.append((mc, pr, "rna"))
+        # dna
+        # if there is only one dna (wes + wgs) PR associated with this MC, pick that PR
+        if (
+            len(
+                prs_in_mc[(prs_in_mc.Datatype == "wgs") | (prs_in_mc.Datatype == "wes")]
+            )
+            == 1
+        ):
+            pr = prs_in_mc[
+                (prs_in_mc.Datatype == "wgs") | (prs_in_mc.Datatype == "wes")
+            ].index[0]
+            rows.append((mc, pr, "dna"))
+        # if there are more than one PRs associated with this MC
+        elif (
+            len(
+                prs_in_mc[(prs_in_mc.Datatype == "wgs") | (prs_in_mc.Datatype == "wes")]
+            )
+            > 1
+        ):
+            cds_ids_wgs = prs_in_mc[prs_in_mc.Datatype == "wgs"].CDSID.tolist()
+            cds_ids_wes = prs_in_mc[prs_in_mc.Datatype == "wes"].CDSID.tolist()
+            pr = ""
+            # if this MC doesn't have any wgs PRs
+            if len(cds_ids_wgs) == 0:
+                # out of the wes PRs, select the PR from the most prioritized source
+                # according to the ranking in source_priority
+                subset_seq_table = seq_table[seq_table.index.isin(cds_ids_wes)]
+                subset_seq_table.source = subset_seq_table.source.replace(
+                    source_priority
+                )
+                latest_cds_id_wes = subset_seq_table.loc[cds_ids_wes, "source"].idxmin()
+                pr = subset_pr_table[subset_pr_table.CDSID == latest_cds_id_wes].index[
+                    0
+                ]
+            # if this MC has wgs PR(s)
+            else:
+                # ignore the wes PRs, select the PR from the most prioritized source
+                # according to the ranking in source_priority from the wgs PR(s)
+                subset_seq_table = seq_table[seq_table.index.isin(cds_ids_wgs)]
+                subset_seq_table.source = subset_seq_table.source.replace(
+                    source_priority
+                )
+                latest_cds_id_wgs = subset_seq_table.loc[cds_ids_wgs, "source"].idxmin()
+                pr = subset_pr_table[subset_pr_table.CDSID == latest_cds_id_wgs].index[
+                    0
+                ]
+            rows.append((mc, pr, "dna"))
+    ach_table = pd.DataFrame(rows, columns=colnames)
+
+    return ach_table
+
+
+def makeDefaultModelTable(
```

Comment:
There are some differences between the two that make it a little complicated to merge into one. Also I've explained the prioritization scheme in detail in both functions so it might be clearer to understand if they're separate functions